source/scripting_v3/GTA/ScriptSettings.cs

```diff
@@ -293,5 +455,75 @@ private void AddNewValueInternal(string sectionName, string keyName, string valu
 				_values.Add(sectionName, newKeyAndValuePairs);
 			}
 		}
+
+		/// <summary>
+		/// Gets a value that indicates whether this <see cref="ScriptSettings"/> has the specified section.
+		/// </summary>
+		public bool HasSection(string section)
+			=> _values.ContainsKey(section);
+
+		/// <summary>
+		/// Gets a value that indicates whether this <see cref="ScriptSettings"/> has the specified key at the specified section.
+		/// </summary>
+		public bool HasKey(string sectionName, string keyName)
+			=> _values.TryGetValue(sectionName, out var keyValuePairs) && keyValuePairs.ContainsKey(keyName);
+
+		/// <summary>
+		/// Gets all of the section names this <see cref="ScriptSettings"/> has.
+		/// </summary>
+		public string[] GetAllSectionNames()
+		{
+			var result = new string[_values.Count];
+			_values.Keys.CopyTo(result, 0);
+
+			return result;
+		}
+
+		/// <summary>
+		/// Gets all of the key names at the specified section name this <see cref="ScriptSettings"/> has.
+		/// </summary>
+		/// <param name="sectionName">The section name.</param>
+		public string[] GetAllKeyNames(string sectionName)
+		{
+			if (!_values.TryGetValue(sectionName, out var keyValuePairs))
+			{
+				return Array.Empty<string>();
+			}
+
+			var result = new string[keyValuePairs.Count];
+			keyValuePairs.Keys.CopyTo(result, 0);
+
+			return result;
+		}
+
+		/// <summary>
+		/// Deletes all of the keys of the specified section this <see cref="ScriptSettings"/> has the key.
+		/// </summary>
+		/// <param name="sectionName">The section name.</param>
+		/// <param name="keyName">The name of the key.</param>
+		/// <returns><see langword="true"/> if the <see cref="ScriptSettings"/> contained the specified key at the specified section and deleted the key; otherwise, <see langword="false"/>.</returns>
+		public bool DeleteKey(string sectionName, string keyName)
```

Comment:
I'd rename the `Delete*` to `Remove*` for slightly more consistency with how other container methods are named.

Reply:
Sure.

Speaking of the method names, I'm thinking I should change the `Has*` to `Contains*` as how other collection names are named such as those of `Dictionary`. Is that acceptable?

Reply:
Yeah, I was thinking about those too. We have a lot of `Has*` methods elsewhere in the script API though, so I wasn't sure about it. But probably still nicer :+1:.